### PR TITLE
Feat: elrond family integration

### PR DESCRIPTION
**Pull request by RamyEB**
## Elrond family integration 

- [x] Created a new family 
- [x] Added tests

### CODE REVIEW

**Review by JunichiSugiura — APPROVED** [review]
LGTM

File: packages/core/src/families/elrond/serializer.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import type { ElrondTransaction, RawElrondTransaction } from "./types";
+
+export function serializeElrondTransaction({
+  family,
+  mode,
+  amount,
+  recipient,
+  fees,
+  data,
+  gasLimit,
```
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:11]: better to sort alphabetically

File: packages/core/src/families/elrond/serializer.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}: ElrondTransaction): RawElrondTransaction {
+  return {
+    family,
+    amount: amount.toString(),
+    recipient,
+    mode,
+    fees: fees ? fees.toString() : undefined,
+    data: data ? data.toString("hex") : undefined,
+    gasLimit: gasLimit.toString(),
```
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:20]: better to sort alphabetically

File: packages/core/tests/serializers.spec.ts [hunk 3]
```diff
@@ -522,6 +524,52 @@ describe("serializers.ts", () => {
     });
   });
 
+  describe("elrond", () => {
+    const family = schemaFamilies.enum.elrond;
+
+    it("should succeed to serialize a elrond transaction with resource and duration", () => {
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:530]:
```suggestion
    it("should serialize an Elrond transaction", () => {
```

File: packages/core/tests/serializers.spec.ts [hunk 4]
```diff
@@ -522,6 +524,52 @@ describe("serializers.ts", () => {
...
+        family,
+        mode: "send",
+        amount: "100",
+        recipient: "recipient",
+        fees: "0",
+        gasLimit: undefined,
+        data: undefined,
+      });
+    });
+  });
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:571]: remove dup test

File: packages/core/tests/serializers.spec.ts [hunk 5]
```diff
@@ -979,6 +1027,46 @@ describe("serializers.ts", () => {
...
+  describe("elrond", () => {
+    const family = schemaFamilies.enum.elrond;
+
+    it("should succeed to deserialize a elrond transaction with resource and duration", () => {
+      const serializedTransaction: RawElrondTransaction = {
+        family,
+        mode: "send",
+        amount: "100",
+        recipient: "recipient",
+      };
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:1039]: add `gasLimit` as it's required.

File: packages/core/tests/serializers.spec.ts [hunk 6]
```diff
@@ -1087,6 +1115,32 @@ describe("serializers.ts", () => {
     });
   });
 
+  describe("elrond", () => {
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:1118]: remove.

File: packages/core/tests/serializers.spec.ts [hunk 7]
```diff
@@ -578,6 +580,32 @@ describe("serializers.ts", () => {
...
+        data: undefined,
+        gasLimit: 0,
+        family,
+        fees: "0",
+        mode: "send",
+        recipient: "recipient",
+      });
+    });
+  });
+
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:608]: move into the describe block of `describe("#serializeTransaction", () => {`

File: packages/core/tests/serializers.spec.ts [hunk 8]
```diff
@@ -1087,6 +1115,32 @@ describe("serializers.ts", () => {
...
+        family,
+        mode: "send",
+        recipient: "recipient",
+        gasLimit: 0,
+        fees: undefined,
+        data: undefined,
+      });
+    });
+  });
+
```
Comment (cksanders) [inline packages/core/tests/serializers.spec.ts:1143]: move into the describe block of describe("#deserializeTransaction", () => {

File: packages/core/src/families/elrond/serializer.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  mode,
+  recipient,
+}: RawElrondTransaction): ElrondTransaction {
+  return {
+    family,
+    mode,
+    amount: new BigNumber(amount),
+    recipient,
+    fees: fees ? new BigNumber(fees) : undefined,
+    data: data ? Buffer.from(data, "hex") : undefined,
```
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:39]: this should be a string

File: packages/core/src/families/elrond/serializer.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  mode,
+  recipient,
+}: ElrondTransaction): RawElrondTransaction {
+  return {
+    family,
+    amount: amount.toString(),
+    recipient,
+    mode,
+    fees: fees ? fees.toString() : undefined,
+    data: data ? data.toString("hex") : undefined,
```
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:19]: should be a string?

File: packages/core/src/families/elrond/serializer.ts (hunk 9 above)
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:39]:
```suggestion
    data,
```

File: packages/core/src/families/elrond/serializer.ts (hunk 10 above)
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:19]:
```suggestion
    data,
```

File: packages/core/src/families/elrond/serializer.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import BigNumber from "bignumber.js";
+import type { ElrondTransaction, RawElrondTransaction } from "./types";
+
+export function serializeElrondTransaction({
+  amount,
+  data,
```
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:6]:
```suggestion
  data = undefined,
```

File: packages/core/src/families/elrond/serializer.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    mode,
+    fees: fees ? fees.toString() : undefined,
+    data: data ? data.toString("hex") : undefined,
+    gasLimit,
+  };
+}
+
+export function deserializeElrondTransaction({
+  amount,
+  data,
```
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:26]:
```suggestion
  data = undefined,
```

File: packages/core/src/families/elrond/serializer.ts (hunk 9 above)
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:39]:
```suggestion
    data: data ? data : undefined,
```

File: packages/core/src/families/elrond/serializer.ts (hunk 10 above)
Comment (cksanders) [inline packages/core/src/families/elrond/serializer.ts:19]:
```suggestion
    data: data ? data : undefined,
```

File: packages/core/src/families/elrond/types.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+import type { TransactionCommon } from "../index";
+import type {
+  schemaRawElrondTransaction,
+  ElrondOperationMode,
+} from "./validation";
+
+export interface ElrondTransaction extends TransactionCommon {
+  readonly family: RawElrondTransaction["family"];
+  mode: ElrondOperationMode;
+  data?: Buffer;
```
Comment (cksanders) [inline packages/core/src/families/elrond/types.ts:12]:
```suggestion
  data?: string;
```

**Pull request merged**